parsl/monitoring/radios/udp_router.py

```diff
@@ -118,6 +116,25 @@ def start(self) -> None:
         finally:
             logger.info("UDP listener finished")
 
+    def process_message(self) -> None:
+        hmdata, addr = self.udp_sock.recvfrom(2048)
+        origin_hmac = hmdata[0:64]
+        data = hmdata[64:]
```

Comment:
64 is such a specific number.  I'm not thinking we'll ever change the digest function, but all the same, I'm questioning the hard-coding of it.  The HMAC instance returns the digest size, so could do something like:

```python
h = hmac.HMAC(key=self.hmac_key, digestmod=self.digest_function)
origin_hmac = hmdata[:h.digest_size]
data = hmdata[h.digest_size:]
```

I don't know if this still uses the C-optimized digest function though.

Reply:
110bfeda8f removes this 64 and also removes a hard coded 128 byte key length (replacing it with a dynamically computer 64 byte key length based on RFC2014 section 2)